lib/record_store/provider/ns1/client.rb

```diff
@@ -10,46 +10,45 @@ def initialize(api_key:)
 
       def zones
         zones = super
-        raise_error(zones) if zones.is_a?(NS1::Response::Error)
+        raise_if_error!(zones)
         zones
       end
 
       def zone(name)
         zone = super(name)
-        raise_error(zone) if zone.is_a?(NS1::Response::Error)
+        raise_if_error!(zone)
         zone
       end
 
       def record(zone:, fqdn:, type:, must_exist: false)
         result = super(zone, fqdn, type)
-        if result.is_a?(NS1::Response::Error)
-          raise_error(result) if must_exist
-          return nil
-        end
+        raise_if_error!(result) if must_exist
+        return nil if result.is_a?(NS1::Response::Error)
         result
       end
 
       def create_record(zone:, fqdn:, type:, params:)
         result = super(zone, fqdn, type, params)
-        raise_error(result) if result.is_a?(NS1::Response::Error)
+        raise_if_error!(result)
         nil
       end
 
       def modify_record(zone:, fqdn:, type:, params:)
         result = super(zone, fqdn, type, params)
-        raise_error(result) if result.is_a?(NS1::Response::Error)
+        raise_if_error!(result)
         nil
       end
 
       def delete_record(zone:, fqdn:, type:)
         result = super(zone, fqdn, type)
-        raise_error(result) if result.is_a?(NS1::Response::Error)
+        raise_if_error!(result)
         nil
       end
 
       private
 
-      def raise_error(result)
+      def raise_if_error!(result)
+        return nil unless result.is_a?(NS1::Response::Error)
```

Comment:
`nil` is not really needed here

```suggestion
        return unless result.is_a?(NS1::Response::Error)
```